hazelcast/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -442,14 +442,13 @@ private void startHeartbeat() {
         long heartbeatInterval = properties.getPositiveMillisOrDefault(HEARTBEAT_INTERVAL);
         ILogger logger = loggingService.getLogger(HeartbeatManager.class);
         HeartbeatManager.start(this, executionService, logger,
-                heartbeatInterval, heartbeatTimeout,
-                Collections.unmodifiableCollection(connectionManager.getActiveConnections()));
+                heartbeatInterval, heartbeatTimeout, connectionManager.getActiveConnections());
     }
 
     private void startIcmpPing() {
         ILogger logger = loggingService.getLogger(HeartbeatManager.class);
         ClientICMPManager.start(config.getNetworkConfig().getClientIcmpPingConfig(), executionService, logger,
-                Collections.unmodifiableCollection(connectionManager.getActiveConnections()));
```

Comment:
why?

Reply:
That was unneeded, we are not modifying the returned collection view